src/views/locale/LocaleStrings.tsx

```diff
@@ -5,26 +5,33 @@ export function getLocaleName(locale: LocaleData): string {
   const languageName = locale.language?.nameDisplay ?? locale.languageCode;
   const territoryName = locale.territory?.nameDisplay ?? locale.territoryCode;
   const scriptName = locale.writingSystem?.nameDisplay ?? locale.explicitScriptCode;
-  const variantName = locale.variantTag?.nameDisplay ?? locale.variantTagCode;
 
-  return (
-    languageName + ' (' + [territoryName, scriptName, variantName].filter(Boolean).join(', ') + ')'
-  );
+  let variantName: string | undefined = undefined;
+  if (locale.variantTags && locale.variantTags.length > 0) {
+    variantName = locale.variantTags.map((v) => v.nameDisplay).join(', ');
+  } else if (locale.variantTagCodes && locale.variantTagCodes.length > 0) {
+    variantName = locale.variantTagCodes.join(', ');
+  }
+
+  return languageName + ' (' + [territoryName, scriptName, variantName].filter(Boolean).join(', ') + ')';
 }
 
 export function getLocaleCode(
   locale: LocaleData,
   localeSeparator: LocaleSeparator,
   territoryOverride?: ISO3166Code,
 ): string {
-  return [
-    locale.language?.codeDisplay ?? locale.languageCode,
-    locale.explicitScriptCode,
-    territoryOverride ?? locale.territoryCode,
-    locale.variantTagCode, // TODO a locale could have multiple variant tags
-  ]
-    .filter(Boolean)
-    .join(localeSeparator);
+  const components: string[] = [];
+ 
+  components.push(locale.language?.codeDisplay ?? locale.languageCode);
+  if (locale.explicitScriptCode) components.push(locale.explicitScriptCode);
+  const territory = territoryOverride ?? locale.territoryCode;
+  if (territory) components.push(territory);
+  // Include all variant subtags
+  if (locale.variantTagCodes && locale.variantTagCodes.length > 0) {
+    components.push(...locale.variantTagCodes);
+  }
+  return components.filter(Boolean).join(localeSeparator);
```

Comment:
It's way easier this way. The push syntax is hard to follow.

```suggestion
 return [
    locale.language?.codeDisplay ?? locale.languageCode,
    locale.explicitScriptCode,
    territoryOverride ?? locale.territoryCode,
    ...locale.variantTagCodes,
  ]
    .filter(Boolean)
    .join(localeSeparator);
```